src/common/users.ts

```diff
@@ -406,7 +408,20 @@ export const clearUserStreak = async (
 // Computes whether we should reset user streak
 // Even though it is the weekend, we should still clear the streak for when the user's last read was Thursday
 // Due to the fact that when Monday comes, we will clear it anyway when we notice the gap in Friday
-export const shouldResetStreak = (day: number, difference: number) => {
+export const shouldResetStreak = (
+  day: number,
+  difference: number,
+  startOfWeek: DayOfWeek = DEFAULT_WEEK_START,
+) => {
+  if (startOfWeek === DayOfWeek.Sunday) {
+    return (
+      (day === Day.Saturday && difference > FREEZE_DAYS_IN_A_WEEK) ||
+      (day === Day.Sunday &&
+        difference > FREEZE_DAYS_IN_A_WEEK + MISSED_LIMIT) ||
+      (day > Day.Sunday && difference > MISSED_LIMIT)
+    );
+  }
```

Comment:
:fml: 😆 a bit hard to parse, but it is what iti s

Reply:
I know 😢
Tried to think of a better way to solve it, but alas to no avail. Maybe worth to take a look another time.

Reply:
@capJavert what do you think about this? A little bit more readable hopefully

```ts
export const shouldResetStreak = (
  day: number,
  difference: number,
  startOfWeek: DayOfWeek = DEFAULT_WEEK_START,
) => {
  const firstDayOfWeek =
    startOfWeek === DayOfWeek.Monday ? Day.Monday : Day.Sunday;

  const lastDayOfWeek =
    startOfWeek === DayOfWeek.Monday ? Day.Sunday : Day.Saturday;

  return (
    (day === lastDayOfWeek && difference > FREEZE_DAYS_IN_A_WEEK) ||
    (day === firstDayOfWeek &&
      difference > FREEZE_DAYS_IN_A_WEEK + MISSED_LIMIT) ||
    (day > firstDayOfWeek && difference > MISSED_LIMIT)
  );
};
```

Reply:
Changed the function to this ☝️

Reply:
Looks better yes, thanks for the effort. I would personally maybe split or expressions in separate if's if we add more down the line, but good for this stage.

Reply:
I wanted to extract them into separate constants, but I couldn't come up with any good names for them :triggered: